inst/design_document.md

```diff
@@ -0,0 +1,241 @@
+the fitted model.
+
+The primary function is `characterize()`, which has methods for
+tidymodels containers (e.g., parsnip or workflow objects) and the
+underlying model fit objects.
+
+When `characterize()` is invoked, an exhaustive list of functions
+(prefixed with `.pluck_`) is executed for the object. The default
+methods for these methods return empty tibbles with appropriate columns.
+If there is a specific method, it is executed, and a tibble is returned
+with a data structure similar to the output produced by the yardstick
+package.
+
+For example:
+
+``` r
+library(tidymodels)
+```
+
+    ── Attaching packages ────────────────────────────────────── tidymodels 1.0.0 ──
+
+    ✔ broom        1.0.1          ✔ recipes      1.0.4     
+    ✔ dials        1.1.0          ✔ rsample      1.1.1.9000
+    ✔ dplyr        1.0.10         ✔ tibble       3.1.8     
+    ✔ ggplot2      3.4.0          ✔ tidyr        1.2.1     
+    ✔ infer        1.0.3          ✔ tune         1.0.1.9002
+    ✔ modeldata    1.0.1.9000     ✔ workflows    1.1.2     
+    ✔ parsnip      1.0.3          ✔ workflowsets 1.0.0     
+    ✔ purrr        1.0.0          ✔ yardstick    1.1.0.9000
+
+    ── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
+    ✖ purrr::discard() masks scales::discard()
+    ✖ dplyr::filter()  masks stats::filter()
+    ✖ dplyr::lag()     masks stats::lag()
+    ✖ recipes::step()  masks stats::step()
+    • Dig deeper into tidy modeling with R at https://www.tmwr.org
+
+``` r
+library(characterize)
+
+tidymodels_prefer()
+theme_set(theme_bw())
+options(pillar.advice = FALSE, pillar.min_title_chars = Inf)
+```
+
+``` r
+# Fit a simple model using parsnip 
+
+cart_fit <- 
+  decision_tree() %>%
+  set_mode("regression") %>% 
+  fit(mpg ~ ., data = mtcars)
+cart_fit
+```
+
+    parsnip model object
+
+    n= 32 
+
+    node), split, n, deviance, yval
+          * denotes terminal node
+
+    1) root 32 1126.04700 20.09062  
+      2) cyl>=5 21  198.47240 16.64762  
+        4) hp>=192.5 7   28.82857 13.41429 *
+        5) hp< 192.5 14   59.87214 18.26429 *
+      3) cyl< 5 11  203.38550 26.66364 *
+
+``` r
+characterize(cart_fit)
+```
+
+    # A tibble: 2 × 3
+      .metric             .estimator .estimate
+      <chr>               <chr>          <dbl>
+    1 num_active_features model              2
+    2 num_term_nodes      model              3
+
+## Sub-models
+
+As with model tuning, there are cases where characteristics can be
+returned for multiple sub-models. For example, glmnet model objects can
+return model coefficients for each value of the `penalty` parameter. For
+such models, a specific `multi_characterize()` method is provided with a
+particular argument for the tuning parameter that indexes the
+sub-models.
+
+## `.pluck` methods
+
+These methods are invoked in `characterize()` and return tibbles with
+columns `statistic` and `value` (string and doubles, respectively).
```

Comment:
Will `characterize()` always return `statistic` and `value` or will it sometimes return a 3rd or 4th column?